### DISCUSSION TITLE
Find selected text

### DISCUSSION THREAD

**Discussion by lua-rocks** [orbitalquark/textadept#124] [Q&A] [answered]
When I select text and press Ctrl+F I would like it to be automatically copied to the search pane. Is it possible?

**Comment by Mirodin** [comment, marked as answer]
```
-- Copy selected text into search field
keys["ctrl+f"] = function()
    local text = not buffer.selection_empty and buffer:get_sel_text() or nil
    ui.find.focus{find_entry_text = text}
end
```

**Reply by lua-rocks** [reply]
Awesome! Thank you.

**Reply by lua-rocks** [reply]
A little improvement:

+ There is no reason to check nils
+ Replacing works inside a selection if it exists, so we need to remove the selection

```lua
keys['ctrl+f'] = function()
  ui.find.focus { find_entry_text = buffer:get_sel_text() }
  buffer:set_selection(0, 0)
end
```

**Reply by Mirodin** [reply]
That is a nice one, thank you. I copied that snippet when I started using Textadept and had not much idea about Lua. Your solution is way more elegant indeed.
